Add basic security functions to druidapi (#14009)
apache/druid: docs/operations/security-overview.md

   druid.auth.authorizer.MyBasicMetadataAuthorizer.type=basic
   ```

3. Restart the cluster.

> See

Reviewer: ```suggestion
See the following topics for more information:
```

* [Authentication and Authorization](../design/auth.md) for more information about the Authenticator,

> Escalator, and Authorizer concepts.

Reviewer: ```suggestion
Escalator, and Authorizer.
```

* [Basic Security](../development/extensions-core/druid-basic-security.md) for more information about
the extension used in the examples above.
* [Kerberos](../development/extensions-core/druid-kerberos.md) for Kerberos authentication.

> * [User authentication and authorization](security-user-auth.md) for details of permissions.

Reviewer: ```suggestion
* [User authentication and authorization](security-user-auth.md) for details about permissions.
```

* [SQL permissions](security-user-auth.md#sql-permissions) for permissions on SQL system tables.

> * The `druidapi` Python library, provided as part of the Druid tutorisls, for functions you can

Reviewer: ```suggestion
* [The `druidapi` Python library](../tutorials/tutorial-jupyter-index.md#python-api-for-druid), provided as part of the Druid tutorials, for functions you can
```

   ```
4. Assign roles to users by issuing a POST request to `druid-ext/basic-security/authorization/db/MyBasicMetadataAuthorizer/users/<USERNAME>/roles/<ROLENAME>`. For example:
	 ```bash
	 curl -u admin:password1 -XPOST https://my-coordinator-ip:8281/druid-ext/basic-security/authorization/db/MyBasicMetadataAuthorizer/users/myname/roles/myrole | jq
	 ```
5. Finally, attach permissions to the roles to control how they can interact with Druid at `druid-ext/basic-security/authorization/db/MyBasicMetadataAuthorizer/roles/<ROLENAME>/permissions`.
	 For example:
	 ```bash
	 curl -u admin:password1 -H'Content-Type: application/json' -XPOST --data-binary @perms.json https://my-coordinator-ip:8281/druid-ext/basic-security/authorization/db/MyBasicMetadataAuthorizer/roles/myrole/permissions
	 ```
	The payload of `perms.json` should be in the form:
 	```json
  [
    {
      "resource": {
        "type": "DATASOURCE",
        "name": "<PATTERN>"
      },
      "action": "READ"
    },
    {
      "resource": {
        "type": "STATE",
        "name": "STATE"
      },
      "action": "READ"
    }
  ]

>   ```

Reviewer 2: I suspect there's a missing backtick in this diff that's causing markdown to render the following section inside a codeblock - please see screenshot:
<img width="1190" alt="CleanShot 2023-04-07 at 09 19 12@2x" src="https://user-images.githubusercontent.com/8687261/230642632-2cb849a2-e56d-4e8f-8cc5-26198376c04f.png">
